cmd/argo/commands/server.go

```diff
@@ -98,10 +97,16 @@ See %s`, help.ArgoServer),
 				managedNamespace = namespace
 			}
 
+			ssoNamespace := namespace
+			if managedNamespace != "" {
+				ssoNamespace = managedNamespace
+			}
```

Comment:
this actually is different from the logic on line 151 and line 160 below. Those set the SSO namespace to the installation namespace when using a managed namespace. That makes sense, since your Argo configuration and Argo Server are in the installation namespace. Only your Workflows are in the managed namespace.

This change is actually quite confusing as a result. And it was an undocumented breaking change that caused a regression too: https://github.com/argoproj/argo-workflows/issues/9989#issuecomment-2077892941.